shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -31,13 +33,6 @@ - (instancetype)init {
   return self;
 }
 
-- (void)dealloc {
-  [_lifeCycleDelegate release];
```

Comment:
`_lifeCycleDelegate` isn't retained in the current MRC init, so it needs to become a weak property.  I wonder if we're just getting away with this because the app delegate is never dealloced before app quit.

Reply:
Err wait; perhaps I'm missing something. `_lifeCycleDelegate` was alloc'ed as an ivar on init and cleaned up in `dealloc`. Why should this be weak?

Reply:
Wait nm I'm confused (as always when I look at this), it's an ivar so it's strongly retained?  Shouldn't a delegate like this be weak?  Is there a retain cycle here?
In any case we should swap the ivar to a property to make the memory semantics clearer.

Reply:
Would be good to add a retain test like this and confirm it passes both before and after your PR.

pseudocode
```objc
__weak id<FlutterApplicationLifeCycleDelegate> weakApplifeCycleDelegate = new real non-mocked object conforming to FlutterApplicationLifeCycleDelegate
@autoreleasepool
appdelegate = FlutterAppDelegate
[appdelegate addApplicationLifeCycleDelegate:weakApplifeCycleDelegate]
appdelegate = nil
@end
XCTAssertNil(weakApplifeCycleDelegate);
```
See https://github.com/flutter/engine/blob/f21f2b232b8a1b0b09c322e5bbe18244519b99e1/shell/platform/darwin/ios/framework/Source/FlutterAppDelegateTest.mm#L139 for example.

Also most of the delegate pass-through code looks untested (addApplicationLifeCycleDelegate for example), would be good to have a test that `UIApplicationDelegate` `-application*` calls are getting passed to the lifecycle delgate.  But that is bonus and I don't think is required for this ARC migration PR.

Reply:
> it's an ivar so it's strongly retained in MRC?

Yep, as written it was strong. 

> Shouldn't a delegate like this be weak?

In this case it's allocated, owned, and collected by FlutterAppDelegate itself, and never publicly accessible since it's in the implementation, so makes sense to be strong here. 

Delegates are typically weak, since they're typically owned by someone else (e.g. handing some object to a table view to look up data from). Delegates also usually are specified by a _protocol_ though... not an actual concrete type like is the case here. 

I think the real issue is that we're calling this a delegate even though it doesn't really follow the usual delegate pattern, where someone else would hand you an object that implements a protocol.

> we should swap the ivar to a property to make the memory semantics clearer

Will do; agreed that'll make this more obvious.

Reply:
You're right, the "delegate" was what was confusing me.  The test I suggested wouldn't add anything over this test, which already exists, since it's just passing from the not-"delegate" protocol to its actual delegate
https://github.com/flutter/engine/blob/f21f2b232b8a1b0b09c322e5bbe18244519b99e1/shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegateTest.mm#L99

Reply:
Done.